### PR TITLE
Fix the locale bug of string convertion.

### PR DESCRIPTION
**Pull request by EternalWind**

### CODE REVIEW

**Comment by svenstaro** [comment]
Apparently you forgot to fix the samples?

File: engine/src/Utils/Utils.hpp [hunk 1]
```diff
@@ -26,6 +27,13 @@ template <typename Source> QString ToString(const Source& source) {
     return QString::fromStdString(boost::lexical_cast<std::string>(source));
 }
 
+/**
+  * Convert a QString to a std::stirng. Use it to replace qt's toStdString() which has a locale bug.
```
Comment (svenstaro) [inline engine/src/Utils/Utils.hpp]: Typo std::stirng

File: engine/src/Utils/Utils.hpp [hunk 2]
```diff
@@ -26,6 +27,13 @@ template <typename Source> QString ToString(const Source& source) {
...
+  * @param qString The QString you are going to convert.
+  * @returns The converted std::string.
+  */
+DUCTTAPE_API std::string ToStdString(const QString& qString);
```
Comment (svenstaro) [inline engine/src/Utils/Utils.hpp]: Please name parameters consistently like qstring, not qString.

File: engine/src/Event/EventListener.hpp [hunk 3]
```diff
@@ -16,6 +16,8 @@
 #include <cstdint>
 #include <memory>
 
+#include <Utils/Utils.hpp>
```
Comment (svenstaro) [inline engine/src/Event/EventListener.hpp]: Local includes first! Read the coding standards.

**Comment by EternalWind** [comment]
That's because there's no toStdString() in the samples.

**Comment by opatut** [comment]
That is right, we use QString everywhere, only when we need to interact with other libraries (Ogre, MyGUI, ...) we convert them. And this should not happen outside the engine :D

**Comment by svenstaro** [comment]
$ grep -r StdString samples/
samples/chat/src/server/Server.cpp:                std::cout << std::endl << c->GetSenderNick().toStdString() << ": " << c->GetMessageText().toStdString() << std::endl;
samples/chat/src/client/Client.cpp:            std::cout << std::endl << "<" << c->GetSenderNick().toStdString() << "> " << c->GetMessageText().toStdString() << std::endl;
samples/chat/src/client/Client.cpp:            QString nick = QString::fromStdString(in.substr(6));
samples/chat/src/client/Client.cpp:            std::cout << "*\* You changed your nick to: " << nick.toStdString() << std::endl;
samples/chat/src/client/Client.cpp:                InjectEvent(std::make_shared<ChatMessageEvent>(QString::fromStdString(in), client->GetNick()));

**Comment by opatut** [comment]
Ah right the chat sample and the std stream :/

**Comment by svenstaro** [comment]
Will merge now and do the remaining changes myself. Thanks for the patch.

**Pull request merged**